x/currencies/handler.go

```diff
@@ -1,30 +1,27 @@
-// Message handler for currencies module.
 package currencies
 
 import (
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	sdkErrors "github.com/cosmos/cosmos-sdk/types/errors"
 
-	"github.com/dfinance/dnode/x/currencies/msgs"
+	"github.com/dfinance/dnode/x/currencies/internal/keeper"
 )
 
-// Handler for currencies messages, provess issue/destroy messages.
-func NewHandler(keeper Keeper) sdk.Handler {
+// NewHandler creates sdk.Msg type messages handler.
+func NewHandler(keeper keeper.Keeper) sdk.Handler {
 	return func(ctx sdk.Context, msg sdk.Msg) (*sdk.Result, error) {
 		switch msg := msg.(type) {
-
-		case msgs.MsgDestroyCurrency:
-			return handleMsgDestroy(ctx, keeper, msg)
-
+		case MsgWithdrawCurrency:
+			return handleMsgWithdraw(ctx, keeper, msg)
 		default:
 			return nil, sdkErrors.Wrapf(sdkErrors.ErrUnknownRequest, "unrecognized currencies msg type: %v", msg.Type())
 		}
 	}
 }
 
-// Handle destroy message.
-func handleMsgDestroy(ctx sdk.Context, keeper Keeper, msg msgs.MsgDestroyCurrency) (*sdk.Result, error) {
-	if err := keeper.DestroyCurrency(ctx, msg.ChainID, msg.Symbol, msg.Recipient, msg.Amount, msg.Spender); err != nil {
+// handleMsgWithdraw handles MsgWithdrawCurrency message.
+func handleMsgWithdraw(ctx sdk.Context, keeper keeper.Keeper, msg MsgWithdrawCurrency) (*sdk.Result, error) {
```

Comment:
could be good to add events later, but currently not so important

Reply:
Task description updated
https://dfin.atlassian.net/browse/DFI-310